sky/backends/cloud_vm_ray_backend.py

```diff
@@ -2684,9 +2684,21 @@ def teardown_no_lock(self,
                         f'gcloud compute tpus tpu-vm list --filter='
                         f'\\(labels.ray-cluster-name={cluster_name}\\) '
                         f'--zone={zone} --format=value\\(name\\)')
-                    terminate_cmd = (
-                        f'gcloud compute tpus tpu-vm delete --zone={zone}'
-                        f' --quiet $({query_cmd})')
+                    returncode, stdout, stderr = log_lib.run_with_log(
+                        query_cmd,
+                        log_abs_path,
+                        shell=True,
+                        stream_logs=False,
+                        require_outputs=True)
+
+                    # Needs to create a list as GCP does not allow deleting
+                    # multiple TPU VMs at once
+                    tpu_terminate_cmds = []
+                    for tpu_id in stdout.splitlines():
+                        tpu_terminate_cmds.append(
+                            f'gcloud compute tpus tpu-vm delete --zone={zone} '
+                            f'--quiet {tpu_id}')
+                    terminate_cmd = ' && '.join(tpu_terminate_cmds)
```

Comment:
We did not handle the first `returncode` here. How about?
```suggestion
                    returncode, stdout, stderr = log_lib.run_with_log(
                        query_cmd,
                        log_abs_path,
                        shell=True,
                        stream_logs=False,
                        require_outputs=True)

                    # Needs to create a list as GCP does not allow deleting
                    # multiple TPU VMs at once
                    # Skip the termination commands, if the TPU ID query commands fail.
                    tpu_terminate_cmds = [f'([[ "{returncode}" == "0" ]] || exit {returncode})']
                    for tpu_id in stdout.splitlines():
                        tpu_terminate_cmds.append(
                            f'gcloud compute tpus tpu-vm delete --zone={zone} '
                            f'--quiet {tpu_id}')
                    terminate_cmd = ' && '.join(tpu_terminate_cmds)
```

Reply:
good point. fixed with minor modification.